common/src/main/java/org/jboss/hal/testsuite/creaper/ResourceVerifier.java

```diff
@@ -0,0 +1,113 @@
+/*
+ * JBoss, Home of Professional Open Source.
+ * Copyright 2015, Red Hat, Inc., and individual contributors
+ * as indicated by the @author tags. See the copyright.txt file in the
+ * distribution for a full listing of individual contributors.
+ *
+ * This is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation; either version 2.1 of
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+
+package org.jboss.hal.testsuite.creaper;
+
+import java.io.IOException;
+
+import org.junit.Assert;
+import org.wildfly.extras.creaper.core.online.ModelNodeResult;
+import org.wildfly.extras.creaper.core.online.OnlineManagementClient;
+import org.wildfly.extras.creaper.core.online.operations.Address;
+import org.wildfly.extras.creaper.core.online.operations.OperationException;
+import org.wildfly.extras.creaper.core.online.operations.Operations;
+
+/**
+ * Created by pjelinek on Nov 16, 2015
+ */
+public class ResourceVerifier {
+
+    private Address resourceAddress;
+    private Operations ops;
+
+    public ResourceVerifier(OnlineManagementClient client) {
+        this.ops = new Operations(client);
+    }
+
+    public ResourceVerifier(Address resourceAddress, OnlineManagementClient client) {
+        this.resourceAddress = resourceAddress;
+        this.ops = new Operations(client);
+    }
+
+    /**
+     * Resource address setter.
+     */
+    public ResourceVerifier forResource(Address address) {
+        this.resourceAddress = address;
+        return this;
+    }
+
+    /**
+     * Verifies if previously set resource exists in model.
+     */
+    public ResourceVerifier verifyResource(boolean expected) throws IOException, OperationException {
+        addressIsSetCheck();
```

Comment:
As you are always expecting address to be set, why it is not a mandatory part of constructor?